libraries/chain/kv_context_rocksdb.cpp

```diff
@@ -233,23 +238,43 @@ namespace eosio { namespace chain {
          temp_data_buffer = nullptr;
 
          std::shared_ptr<const bytes> old_value;
+         int64_t old_value_size = 0;
          try {
             try {
                old_value = view.get(contract, { key, key_size });
-               view.set(contract, { key, key_size }, { value, value_size });
+               if (old_value) {
+                  EOS_ASSERT(static_cast<int64_t>(old_value->size()) >= kv_payer_size, kv_rocksdb_bad_value_size_exception , "The size of value returned from RocksDB is less than payer's size");
+                  old_value_size = static_cast<int64_t>(old_value->size()) - kv_payer_size;
+               }
+
+               // Need to store payer together with value for
```

Comment:
change comment to something like "need to store payer to properly credit this account when storage is removed or changed to another payer"